web/source/osk/anchoredOskView.ts

```diff
@@ -150,17 +150,27 @@ namespace com.keyman.osk {
         return keymanweb['getOskHeight']();
       }
 
-      var oskHeightLandscapeView=Math.floor(Math.min(screen.availHeight,screen.availWidth)/2),
+      let baseWidth  = document?.documentElement?.clientWidth;
+      let baseHeight = document?.documentElement?.clientHeight;
+      if(typeof baseWidth == 'undefined') {
+        baseWidth  = Math.min(screen.height, screen.width);
+        baseHeight = Math.max(screen.height, screen.width);
+
+        if(!keymanweb.util.portraitView()) {
+          let temp = baseWidth;
+          baseWidth = baseHeight;
+          baseHeight = temp;
+        }
+      }
+
+      var oskHeightLandscapeView=Math.floor(Math.min(baseHeight, baseWidth)/2),
           height=oskHeightLandscapeView;
 
       if(device.formFactor == 'phone') {
-        var sx=Math.min(screen.height,screen.width),
-            sy=Math.max(screen.height,screen.width);
-
         if(keymanweb.util.portraitView())
-          height=Math.floor(Math.max(screen.availHeight,screen.availWidth)/3);
+          height=Math.floor(baseHeight/2.4);
```

Comment:
I don't understand why the divisor changed from `3` to `2.4` here?

Reply:
That's fair.

Note the addition of `baseHeight` earlier in the function, which is based on the height of the document element.  This is contained within the browser / WebView and will almost certainly be smaller than the _screen_, which originally was used to provide a base height.

In initial development and testing, keeping the value the same as before produced a very "short" keyboard, notably shrunken compared to before this branch's changes.  This new figure seems to approximate the old height pretty well.

This is also part of why I specified tablet targets for some of the user tests; I wanted to be sure the changes didn't cause uncomfortable side-effects there, as there would naturally be different scale-ratio between document and screen there.

Reply:
Cool bananas. Makes sense now. Can you document that above this line as a code comment for future maintainers?